data-model/src/main/java/org/hypertrace/core/datamodel/shared/StructuredTraceGraph.java

```diff
@@ -48,40 +46,52 @@ private StructuredTraceGraph() {
    */
   public static StructuredTraceGraph createGraph(StructuredTrace trace) {
     StructuredTraceGraph graph = new StructuredTraceGraph();
+    graph.buildEventMap(trace);
+    graph.buildEntityMap(trace);
     graph.buildParentChildRelationship(trace);
     graph.buildRootEvents(trace);
     graph.buildRootEntities(trace);
     return graph;
   }
 
   public Set<Event> getRootEvents() {
-    return ImmutableSet.copyOf(rootEvents);
+    return rootEvents;
   }
 
   public Set<Entity> getRootEntities() {
```

Comment:
Maybe javadoc can be added saying that the returned set is immutable

Reply:
Sure, added javadocs